ipa-core/tests/hybrid.rs

```diff
@@ -134,3 +146,213 @@ fn test_hybrid() {
+    command
+        .arg("serve-input")
+        .preserved_fds(vec![poll_port.as_raw_fd()])
+        .args(["--fd", &poll_port.as_raw_fd().to_string()])
+        .args([
+            "--dir".as_ref(),
+            upload_metadata.parent().unwrap().as_os_str(),
+        ])
+        .silent();
+
+    let _server_handle = command.spawn().unwrap();
+
+    // Run Hybrid
+    let mut command = Command::new(TEST_RC_BIN);
+    command
+        .args(["--network".into(), config_path.join("network.toml")])
+        .args(["--output-file".as_ref(), output_file.as_os_str()])
+        .args(["--shard-count", SHARDS.to_string().as_str()])
+        .args(["--wait", "2"])
+        .arg("malicious-hybrid")
+        .silent()
+        .args(["--count", INPUT_SIZE.to_string().as_str()])
+        .args(["--url-file-list".into(), upload_metadata])
+        .args(["--max-breakdown-key", &config.max_breakdown_key.to_string()]);
+
+    match config.with_dp {
+        0 => {
+            command.args(["--with-dp", &config.with_dp.to_string()]);
+        }
+        _ => {
+            command
+                .args(["--with-dp", &config.with_dp.to_string()])
+                .args(["--epsilon", &config.epsilon.to_string()]);
+        }
+    }
+    command.stdin(Stdio::piped());
+
+    let test_mpc = command.spawn().unwrap().terminate_on_drop();
+    test_mpc.wait().unwrap_status();
+
+    // basic output checks - output should have the exact size as number of breakdowns
+    let output = serde_json::from_str::<HybridQueryResult>(
+        &std::fs::read_to_string(&output_file).expect("IPA results file should exist"),
+    )
+    .expect("IPA results file is valid JSON");
+
+    assert_eq!(
+        usize::try_from(config.max_breakdown_key).unwrap(),
+        output.breakdowns.len(),
+        "Number of breakdowns does not match the expected",
+    );
+    assert_eq!(INPUT_SIZE, usize::from(output.input_size));
+
+    let expected_result: Vec<u32> = from_reader(
+        File::open(in_the_clear_output_file)
+            .expect("file should exist as it's created above in the test"),
+    )
+    .expect("should match hard coded format from in_the_clear");
+    assert!(output
+        .breakdowns
+        .iter()
+        .zip(expected_result.iter())
+        .all(|(a, b)| a == b));
+}
+
+fn create_upload_files<const SHARDS: usize>(
+    enc_file1: &Path,
+    enc_file2: &Path,
+    enc_file3: &Path,
+    port: u16,
+    dest: &Path,
+) -> Result<PathBuf, BoxError> {
+    let manifest_path = dest.join("manifest.txt");
+    let mut manifest_file = File::create_new(&manifest_path)?;
+    create_upload_file::<SHARDS>("h1", enc_file1, port, dest, &mut manifest_file)?;
+    create_upload_file::<SHARDS>("h2", enc_file2, port, dest, &mut manifest_file)?;
+    create_upload_file::<SHARDS>("h3", enc_file3, port, dest, &mut manifest_file)?;
+
+    manifest_file.flush()?;
+
+    Ok(manifest_path)
+}
+
+fn create_upload_file<const SHARDS: usize>(
```

Comment:
this works, but my original plan was to generate `SHARDS` raw input files, then encrypt each with `--length-delimited`.  either way is fine with me, this just puts a little extra weight on the rust implementation.

Reply:
how would you run ipa in the clear on those?